scripts/player.gd

```diff
@@ -54,5 +63,7 @@ func set_coyote_jump_enabled(value: bool) -> void:
 func is_coyote_jump_enabled() -> bool:
 	return !$CoyoteJumpTimer.is_stopped()
 
-func player_die() -> void:
-	queue_free()
+
+func die() -> void:
+	# TODO: transition to die state
```

Comment:
Any idea what the death function call or state would look like? Would have to change the logic in the hitbox.tscn that calls the player's death

Reply:
just call body.die() from Enemy, Player will handle the rest.
Player will probably transition to DeathState where an animation plays, then signal the Game to transition back to the same level (recreate the same level again)

Reply:
sounds good.